usbSerialExamples/src/main/java/com/mikey0000/android/usbserial/examples/NMEAParser.java

```diff
@@ -94,32 +93,44 @@ public boolean parse(String[] tokens, GPSPosition position) {
         public float time = 0.0f;
         public float lat = 0.0f;
         public float lon = 0.0f;
-        public boolean fixed = false;
         public int quality = 0;
         public float dir = 0.0f;
         public float altitude = 0.0f;
         public float velocity = 0.0f;
+        public int numberOfSatelites = 0;
+        public float hdop = 0.0f;
+
+        public boolean fixed = false;
 
         public void updatefix() {
             fixed = quality > 0;
         }
 
         public String toString() {
-            return String.format("POSITION: lat: %f, lon: %f, time: %f, Q: %d, dir: %f, alt: %f, vel: %f", lat, lon, time, quality, dir, altitude, velocity);
+            return String.format(
+                    "POSITION: lat: %f, lon: %f, time: %f, Q: %d, hdop: %f, dir: %f, alt: %f, vel: %f",
+                    lat, lon, time, quality, hdop, dir, altitude, velocity
+            );
         }
     }
 
     GPSPosition position = new GPSPosition();
 
     private static final Map<String, SentenceParser> sentenceParsers = new HashMap<String, SentenceParser>();
+    protected static final Map<Integer,Float> qualityAccuracy = new HashMap<Integer, Float>();
 
     public NMEAParser() {
         sentenceParsers.put("GPGGA", new GPGGA());
-        sentenceParsers.put("GPGGL", new GPGGL());
-        sentenceParsers.put("GPRMC", new GPRMC());
-        sentenceParsers.put("GPRMZ", new GPRMZ());
+        //sentenceParsers.put("GPGGL", new GPGGL());
+        //sentenceParsers.put("GPRMC", new GPRMC());
+        //sentenceParsers.put("GPRMZ", new GPRMZ());
```

Comment:
I re-enabled those NMEA sentences, but I think we can skip the parsing because only a position with a 'quality' value gets pushed to the mock location provider. Those commented-out sentences are not setting the quality, so every thing from those sentences gets overwritten by a 'GPGGA' sentence.